### PR TITLE
194 refactor relavant fields

### PR DESCRIPTION
**Pull request by mhughes27**

### CODE REVIEW

File: app/views/admin/awards/_form.html.erb [hunk 1]
```diff
@@ -30,11 +30,14 @@ and renders all form fields for a resource's editable attributes.
...
-  <% page.attributes.each do |attribute| -%>
-    <div class="field-unit field-unit--<%= attribute.html_class %>">
-      <%= render_field attribute, f: f %>
-    </div>
-  <% end -%>
+  <div class="">
+    <%= f.label :award_name %>
+    <%= f.text_field :award_name %><br>
+    <%= f.label :award_type %>
+    <%=f.select(:award_type, options_for_select([['ACTING', 'acting'],['DIRECTING','directing'],['TECHNICAL','technical'],['PRODUCTION','production'],['ENSEMBLE','ensemble']])) %><br>
```
Comment (sumeetjain) [inline app/views/admin/awards/_form.html.erb]: Move code into helper (array_count >= 3)

File: db/migrate/20160812154427_remove_relevant_fields_from_awards.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class RemoveRelevantFieldsFromAwards < ActiveRecord::Migration
+  def change
+    remove_column :awards, :relevant_fields, :integer
+
```
Comment (sumeetjain) [inline db/migrate/20160812154427_remove_relevant_fields_from_awards.rb]: Remove trailing whitespace

File: app/views/admin/awards/_form.html.erb [hunk 3]
```diff
@@ -30,11 +30,14 @@ and renders all form fields for a resource's editable attributes.
...
+    <%= f.label :inactive %>
```
Comment (sumeetjain) [inline app/views/admin/awards/_form.html.erb]: Remove trailing whitespace

File: app/views/admin/awards/new.html.erb [hunk 4]
```diff
@@ -24,6 +24,18 @@ to do the heavy lifting.
...
 
-<div>For award categories that only include a name, type '1' in 'relevant fields'. For awards that will include a theater and a show, enter '2' in 'relevant fields'. If an award will additionally include a nominee field but not a role, enter '3'. If the award will additionally include both a nominee and a role field, enter '4'.</div>
+<div>Award type is only used to populate the correct input fields for the nomination form.  Below is a list of input fields that populate with each type.<br>
+<br>
+
+ACTING: Theater, Show, Nominee, Role<br>
+DIRECTING: Theater, Show, Nominee<br>
+TECHNICAL: Theater, Show, Nominee<br>
+PRODUCTION: Theater, Show<br>
+ENSEMBLE: Theater, Show<br>
```
Comment (sumeetjain) [inline app/views/admin/awards/new.html.erb]: Remove trailing whitespace

File: db/migrate/20160811184502_add_award_type_to_awards.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddAwardTypeToAwards < ActiveRecord::Migration
```
Comment (sumeetjain) [inline db/migrate/20160811184502_add_award_type_to_awards.rb]: Has no descriptive comment (IrresponsibleModule)

File: db/migrate/20160811190725_add_award_type_to_award.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddAwardTypeToAward < ActiveRecord::Migration
```
Comment (sumeetjain) [inline db/migrate/20160811190725_add_award_type_to_award.rb]: Has no descriptive comment (IrresponsibleModule)

File: db/migrate/20160812154427_remove_relevant_fields_from_awards.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class RemoveRelevantFieldsFromAwards < ActiveRecord::Migration
```
Comment (sumeetjain) [inline db/migrate/20160812154427_remove_relevant_fields_from_awards.rb]: Has no descriptive comment (IrresponsibleModule)

File: app/dashboards/award_dashboard.rb [hunk 8]
```diff
@@ -10,11 +10,12 @@ class AwardDashboard < Administrate::BaseDashboard
   ATTRIBUTE_TYPES = {
     id: Field::Number,
     award_name: Field::String,
-    relevant_fields: Field::Number,
+    # relevant_fields: Field::Number,
     created_at: Field::DateTime,
     updated_at: Field::DateTime,
     inactive: Field::Boolean,
     nominations: Field::HasMany,
+    award_type: Field::String,
```
Comment (sumeetjain) [inline app/dashboards/award_dashboard.rb]: Avoid comma after the last item of a hash.

File: app/dashboards/award_dashboard.rb [hunk 9]
```diff
@@ -24,14 +25,16 @@ class AwardDashboard < Administrate::BaseDashboard
   # Feel free to add, remove, or rearrange items.
   COLLECTION_ATTRIBUTES = [
     :award_name,
-    :relevant_fields,
+    # :relevant_fields,
+    :award_type,
```
Comment (sumeetjain) [inline app/dashboards/award_dashboard.rb]: Avoid comma after the last item of an array.

File: db/migrate/20160811184502_add_award_type_to_awards.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
...
+  def change
+
```
Comment (sumeetjain) [inline db/migrate/20160811184502_add_award_type_to_awards.rb]: Trailing whitespace detected.

Comment (sumeetjain) [inline db/migrate/20160811184502_add_award_type_to_awards.rb]: Tab detected.

File: db/migrate/20160812154427_remove_relevant_fields_from_awards.rb (hunk 2 above)
Comment (sumeetjain) [inline db/migrate/20160812154427_remove_relevant_fields_from_awards.rb]: Trailing whitespace detected.

File: db/schema.rb [hunk 11]
```diff
@@ -11,18 +11,18 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20160812144450) do
+ActiveRecord::Schema.define(version: 20160812154427) do
```
Comment (sumeetjain) [inline db/schema.rb:14]: Separate every 3 digits in the integer portion of a number with underscores(_).

**Comment by sumeetjain** [comment]
- [x] This work will break the currently deployed application, as none of the awards that exist will have a value for `award_type`. Add to the DB migration script some code that sets each award's `award_type`.
- [x] If something isn't needed anymore, remove it. Don't comment it out, e.g. `# relevant_fields: Field::Number,`
- [x] In this file - https://github.com/omahacodeschool/tag-awards/pull/199/files#diff-4c2574d06b455fb92a892e0469893a24 - I see why you removed the automatic generation of form fields, but that means that fields for future additions to the Award model will have to be manually added. It negates the purpose of the gem for automatic form building. Instead, leave what was there previously but just add `<%=f.select(:award_type, options_for_select([['ACTING', 'acting'],['DIRECTING','directing'],['TECHNICAL','technical'],['PRODUCTION','production'],['ENSEMBLE','ensemble']])) %>` below the automatically generated fields. Or look into defining custom field types in Adminstrate (the gem we're using for the administration area)
- [ ] Extract `[['ACTING', 'acting'],['DIRECTING','directing'],['TECHNICAL','technical'],['PRODUCTION','production'],['ENSEMBLE','ensemble']]` into a class method in `Award`, so that it's easy to re-use that data structure elsewhere if needed.
- [x] In **app/views/nominations/nomination_ballot.html.erb**, theater and show are _always_ required. You can refactor to DRY up that view a bit by using the conditionals to just add the field that are unique to a particular kind of award.

File: app/views/admin/awards/_form.html.erb [hunk 12]
```diff
@@ -36,7 +36,12 @@ and renders all form fields for a resource's editable attributes.
     </div>
   <% end -%>
 
-  <div class="form-actions">
-    <%= f.submit %>
-  </div>
+    <div class="field-unit">
+      <%= f.label :award_type %>
+      <%=f.select(:award_type, options_for_select([['ACTING', 'acting'],['DIRECTING','directing'],['TECHNICAL','technical'],['PRODUCTION','production'],['ENSEMBLE','ensemble']])) %>
```
Comment (sumeetjain) [inline app/views/admin/awards/_form.html.erb:41]: Move code into helper (array_count >= 3)

File: db/migrate/20160811184502_add_award_type_to_awards.rb [hunk 13]
```diff
@@ -0,0 +1,4 @@
+class AddAwardTypeToAwards < ActiveRecord::Migration
```
Comment (sumeetjain) [inline db/migrate/20160811184502_add_award_type_to_awards.rb:1]: Has no descriptive comment (IrresponsibleModule)

File: db/migrate/20160811190725_add_award_type_to_award.rb (hunk 6 above)
Comment (sumeetjain) [inline db/migrate/20160811190725_add_award_type_to_award.rb:1]: Has no descriptive comment (IrresponsibleModule)

File: db/migrate/20160812154427_remove_relevant_fields_from_awards.rb [hunk 14]
```diff
@@ -0,0 +1,5 @@
...
```
Comment (sumeetjain) [inline db/migrate/20160812154427_remove_relevant_fields_from_awards.rb:1]: Has no descriptive comment (IrresponsibleModule)

File: app/views/nominations/nomination_ballot.html.erb [hunk 15]
```diff
@@ -37,15 +37,17 @@
...
+                <% if award.award_type != ""  %>
                   Theater:  <input class="theater_input" type="text" name="awards[<%= award.id %>][nominations[<%= nomination_count %>][theater]]" value="<%= current_user.nomination_value(award.id, nomination_count, "theater")%>">
-                  Show:  <input type="text" name="awards[<%= award.id %>][nominations[<%= nomination_count %>][show]]" value="<%= current_user.nomination_value(award.id, nomination_count, "show") %>">
+                  Show:  <input type="text" name="awards[<%= award.id %>][nominations[<%= nomination_count %>][show]]" value="<%= current_user.nomination_value(award.id, nomination_count, "show") %>"><br>
                 <%end%>
-                <% if award.relevant_fields >= 3 || award.relevant_fields == 1 %>
-                  <br>
-                  Nominee:  <input class="theater_input" type="text" name="awards[<%= award.id %>][nominations[<%= nomination_count %>][nominee]]" value="<%= current_user.nomination_value(award.id, nomination_count, "nominee") %>">
+                
+                <% if award.award_type == "directing" || award.award_type == "technical" || award.award_type == "acting" %>
```
Comment (sumeetjain) [inline app/views/nominations/nomination_ballot.html.erb]: Move code into model (award use_count > 2)

**Comment by mhughes27** [comment]
When I added the needs_nominee_input? method to the Award model I get an undefined local variable or method error.  What am I missing?  I have push my latest code.

File: app/models/award.rb [hunk 16]
```diff
@@ -12,4 +12,9 @@ def short_name
 
     award_name.gsub(regex, "")
   end
+
+  def needs_nominee_input?
+    if award.award_type == "directing" || award.award_type == "technical" || award.award_type == "acting"
```
Comment (sumeetjain) [inline app/models/award.rb]: Calls award.award_type 3 times (DuplicateMethodCall)

Comment (sumeetjain) [inline app/models/award.rb]: Line is too long. [105/80]

**Comment by mhughes27** [comment]
I did it a little differently.  Is my way ok

File: app/models/award.rb [hunk 17]
```diff
@@ -12,4 +12,23 @@ def short_name
 
     award_name.gsub(regex, "")
   end
+
+  def needs_nominee_input?
+    type = award_type
+    if type == "directing" || type == "technical" || type == "acting"
```
Comment (sumeetjain) [inline app/models/award.rb]: Refers to type more than self (maybe move it to another class?) (FeatureEnvy)

File: app/models/award.rb [hunk 18]
```diff
@@ -12,4 +12,23 @@ def short_name
...
+      return true
+    else
+      return false
+    end
+  end
+
+  def needs_role_input?
+    type = award_type
+    if type == "acting"
```
Comment (sumeetjain) [inline app/models/award.rb]: Use a guard clause instead of wrapping the code inside a conditional expression.

File: app/models/award.rb [hunk 19]
```diff
@@ -12,4 +12,23 @@ def short_name
...
+      return true
+    else
+      return false
+    end
+  end
+
```
Comment (sumeetjain) [inline app/models/award.rb:23]: Extra empty line detected at class body end.

**Comment by sumeetjain** [comment]
It does work, but it's redundant to declare `type = award_type` and then use `type` in the exact way that `award_type` could have been used. So you should not have `type = award_type` at all.

`self` **is** optional, in fact, as Ruby will assume `self`. So the following should work:

``` ruby
if award_type == "directing" || award_type == "technical" || award_type == "acting"
  return true
else
  return false
end
```

In fact, that can be further shortened to this:

``` ruby
award_type == "directing" || award_type == "technical" || award_type == "acting"
```

Because that single expression _inherently returns either True or False_ already. Try it out!

**Pull request merged**